### ISSUE TITLE
Initialization step in DMM

### ISSUE THREAD

**Issue by dgeng7** [riccardomiotto/py-dmm#1]
Initializing the same alpha for all mixture components will lead to the same alpha for all the components in the end.

**Comment by riccardomiotto** [comment]
If you run the code, you see that the final alphas are still different across the components. You can change the initialization step and make it random though in order to obtain alpha coefficients more different.

**Comment by dgeng7** [comment]
This is what I did and got:

from mixturemodel import DMM
import pandas as pd

ratio = pd.read_csv("ratio_sample.csv")
dmm  = DMM()
dmm.training(ratio.values,maxiter = 100)
dmm.alpha
array([[0.24913083, 0.09688852, 0.11200627, 0.61390885, 0.10281855],
       [0.24913083, 0.09688852, 0.11200627, 0.61390885, 0.10281855],
       [0.24913083, 0.09688852, 0.11200627, 0.61390885, 0.10281855],
       [0.24913083, 0.09688852, 0.11200627, 0.61390885, 0.10281855]])

All components are shown as converged.

**Comment by dgeng7** [comment]
I think when initializing the same alpha across components, dmm becomes a single component dirichlet.

**Comment by riccardomiotto** [comment]
I don't remember having had this problem (the code is maybe 8 years old); maybe change the initialization step and make it random

**Comment by dgeng7** [comment]
yeah, making it random will work. initializing using kmeans might be a better alternative. Thank you so much again for providing the code!

**Comment by riccardomiotto** [comment]
Thanks for spotting the issue; feel free to push the new initialization if it works better (just added you as collaborator).
